test/test_alias.cpp

```diff
@@ -263,4 +263,95 @@ TEST_F(AliasTest, ViewPermute) {
   testValidate(fec.fusion(), {out_tensor}, {in_tensor}, __LINE__, __FILE__);
 }
 
+TEST_F(AliasTest, DuplicateOutputs) {
+  {
+    // testing a complete fusion
+    auto fusion = std::make_unique<Fusion>();
+    FusionGuard fg(fusion.get());
+
+    const std::vector<int64_t> in_shape({2, 3, 4});
+
+    TensorView* in = makeContigConcreteTensor(in_shape);
+    fusion->addInput(in);
+    TensorView* out = add(in, IrBuilder::create<Val>(3.141));
+    fusion->addOutput(out);
+    fusion->addOutput(out); // duplicated outputs
+
+    FusionExecutorCache fec(std::move(fusion));
+    at::Tensor in_tensor =
+        at::randn(in_shape, at::dtype(at::kFloat).device(at::kCUDA, 0));
+    std::vector<at::Tensor> out_tensors = fec.runFusionWithInputs({in_tensor});
+    ASSERT_EQ(out_tensors.size(), 2);
+    at::Tensor out_tensor_0 = out_tensors[0];
+    at::Tensor out_tensor_1 = out_tensors[1];
+
+    // Verify aliasing among duplicated outputs
+    EXPECT_TRUE(out_tensor_0.is_alias_of(out_tensor_1));
+    // Verify no segmentation
+    NVF_CHECK(
+        !fec.getMostRecentKernelRuntime()->isSegmented(),
+        "segmentation is not supposed to happen");
+
+    at::Tensor out_tensor = in_tensor.add(3.141);
+    // Verify output values.
+    testValidate(
+        fec.fusion(),
+        {out_tensor, out_tensor},
+        {in_tensor},
+        __LINE__,
+        __FILE__);
+  }
+
+  {
+    // testing duplicated output in segmented fusion
```

Comment:
Can this be a separate test? It looks very independent and there are many benefits from small tests in general, e.g., isolation and parallelization.